src/profile/components/saveIndicator/SaveIndicator.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import commonFormStyles from '../../../common/cssHelpers/form.module.css';
+import { Action } from '../../hooks/useProfileDataEditor';
+
+type Props = {
+  action: Action | undefined;
+  testId: string;
+};
+
+const SaveIndicator = (props: Props): React.ReactElement | null => {
+  const { action, testId } = props;
+  const { t } = useTranslation();
+  if (!action) {
```

Comment:
I would prefer this check to be in BasicData.tsx, but it can be just a preference thing. If you render the SaveIndicator component now, you can't be sure that it actually renders what it says, but it can return null also.

Reply:
Same comment than above: components that are used by multiple components, should make these check internally in once place instead of in all components that use it and it's ok to return null.

https://github.com/City-of-Helsinki/open-city-profile-ui/pull/154#discussion_r617231848